File: shared/serialization/src/commonTest/kotlin/kmp/benchmarks/serialization/Resources.kt
package kmp.benchmarks.serialization

import com.goncalossilva.resources.Resource

object Resources {
    object MinimisedJsonString {
        val largeList = Resource("src/commonTest/resources/large-list-min.json").readText()
        val macosReleases = Resource("src/commonTest/resources/macos-releases-min.json").readText()
        val polymorphicGeo = Resource("src/commonTest/resources/polymorphic-geo-min.json").readText()
        val polymorphicHtml = Resource("src/commonTest/resources/polymorphic-html-min.json").readText()
        val userProfile = Resource("src/commonTest/resources/user-profile-min.json").readText()
    }

    object PrettyPrintedJsonString {
        val largeList = Resource("src/commonTest/resources/large-list.json").readText()
        val macosReleases = Resource("src/commonTest/resources/macos-releases.json").readText()
        val polymorphicGeo = Resource("src/commonTest/resources/polymorphic-geo.json").readText()
        val polymorphicHtml = Resource("src/commonTest/resources/polymorphic-html.json").readText()
        val userProfile = Resource("src/commonTest/resources/user-profile.json").readText()
    }
}

File: app-ios/standaloneTests/serialization/Resources.swift
//
//  Resources.swift
//  standaloneTests
//
//  Created by Sebastian Owodzin on 16/01/2024.
//  Copyright © 2024 kmp. All rights reserved.
//

import Foundation

extension Bundle {
    public static var testResources: Bundle { Resources().bundle }
}

public class Resources {
    internal var bundle: Bundle {
        return Bundle(for: type(of: self))
    }
    
    public struct Minimised {
        public static var largeListJsonString: String = try! String(contentsOfFile: Bundle.testResources.path(forResource: "large-list-min", ofType: "json")!)
        
        public static var macosReleasesJsonString: String = try! String(contentsOfFile: Bundle.testResources.path(forResource: "macos-releases-min", ofType: "json")!)
        
        public static var polymorphicGeoJsonString: String = try! String(contentsOfFile: Bundle.testResources.path(forResource: "polymorphic-geo-min", ofType: "json")!)
        
        public static var polymorphicHtmlJsonString: String = try! String(contentsOfFile: Bundle.testResources.path(forResource: "polymorphic-html-min", ofType: "json")!)
        
        public static var userProfileJsonString: String = try! String(contentsOfFile: Bundle.testResources.path(forResource: "user-profile-min", ofType: "json")!)
    }
    
    public struct PrettyPrinted {
        public static var largeListJsonString: String = try! String(contentsOfFile: Bundle.testResources.path(forResource: "large-list", ofType: "json")!)
        
        public static var macosReleasesJsonString: String = try! String(contentsOfFile: Bundle.testResources.path(forResource: "macos-releases", ofType: "json")!)
        
        public static var polymorphicGeoJsonString: String = try! String(contentsOfFile: Bundle.testResources.path(forResource: "polymorphic-geo", ofType: "json")!)
        
        public static var polymorphicHtmlJsonString: String = try! String(contentsOfFile: Bundle.testResources.path(forResource: "polymorphic-html", ofType: "json")!)
        
        public static var userProfileJsonString: String = try! String(contentsOfFile: Bundle.testResources.path(forResource: "user-profile", ofType: "json")!)
    }
}
